app/frontend/src/features/profile/ProfileTagInput.tsx

```diff
@@ -0,0 +1,223 @@
+import React, { useState, useEffect } from "react";
+import {
+  Checkbox,
+  IconButton,
+  InputBase,
+  ButtonBase,
+  Popper,
+  fade,
+  makeStyles,
+  createStyles,
+} from "@material-ui/core";
+import Autocomplete, {
+  AutocompleteCloseReason,
+} from "@material-ui/lab/Autocomplete";
+import { CloseIcon, ExpandMore } from "../../components/Icons";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    popper: {
+      border: "1px solid rgba(27,31,35,.15)",
+      boxShadow: "0 3px 12px rgba(27,31,35,.15)",
+      borderRadius: 3,
+      zIndex: 1,
+      color: "#586069",
+      backgroundColor: "#f6f8fa",
+    },
+    button: {
+      borderRadius: theme.shape.borderRadius,
+      borderColor: "rgba(0, 0, 0, 0.23)",
+      borderWidth: 1,
+      borderStyle: "solid",
+      padding: "18.5px 14px",
+      margin: theme.spacing(1, 0),
+      width: "100%",
+      fontSize: "16px",
+      justifyContent: "space-between",
+    },
+    tag: {
+      padding: "0 14px",
+      margin: theme.spacing(1, 0),
+      fontSize: "16px",
+      display: "flex",
+      alignItems: "center",
+    },
+    tagLabel: {
+      marginLeft: theme.spacing(1),
+    },
+    header: {
+      borderBottom: "1px solid #e1e4e8",
+      padding: "8px 10px",
+      fontWeight: 600,
+    },
+    inputBase: {
+      padding: 10,
+      width: "100%",
+      borderBottom: "1px solid #dfe2e5",
+      "& input": {
+        borderRadius: 4,
+        backgroundColor: theme.palette.common.white,
+        padding: 8,
+        transition: theme.transitions.create(["border-color", "box-shadow"]),
+        border: "1px solid #ced4da",
+        fontSize: 14,
+        "&:focus": {
+          boxShadow: `${fade(theme.palette.primary.main, 0.25)} 0 0 0 0.2rem`,
+          borderColor: theme.palette.primary.main,
+        },
+      },
+    },
+    paper: {
+      boxShadow: "none",
+      margin: 0,
+      color: "#586069",
+    },
+    option: {
+      minHeight: "auto",
+      alignItems: "flex-start",
+      padding: 8,
+      '&[aria-selected="true"]': {
+        backgroundColor: "transparent",
+      },
+      '&[data-focus="true"]': {
+        backgroundColor: theme.palette.action.hover,
+      },
+    },
+    popperDisablePortal: {
+      position: "relative",
+    },
+    checkbox: {
+      padding: 0,
+      marginRight: theme.spacing(1),
+    },
+  })
+);
+
+interface ProfileTagInputProps {
+  onChange: (_: any, value: string[]) => void;
+  value: string[];
+  options: string[];
+  label: string;
+}
+
+export default function ProfileTagInput({
+  onChange,
+  value,
+  options,
+  label,
+}: ProfileTagInputProps) {
+  const classes = useStyles();
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
```

Comment:
```suggestion
  const anchorElRef = useRef<HTMLElement>();
```

Reply:
I can use `useRef` here but it would mean adding a separate `useState` to keep track of if the popper is open. Right now, the `anchorEl` state is doing both. The Popper component will also only accept `anchorEl.current` if using `useRef`. 

Is it better in this case to separate state/ref and use `useRef`? For readability?

Reply:
Yes, it's shorter now but having ref and open state separate is more immediately clear if someone else is working with this code later.